Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -318,26 +322,46 @@ namespace ROS2::Utils
             }
 
             AZ::IO::Path resolvedPath(resolved);
+            const bool needsVisual = visuals.contains(unresolvedUrfFileName);
             const bool needsCollider = colliders.contains(unresolvedUrfFileName);
 
             const AZStd::string pathSuffix = needsCollider ? "_collider" : "";
             const AZStd::string newFilename = resolvedPath.Stem().String() + pathSuffix + resolvedPath.Extension().String();
             AZ::IO::Path targetPathAssetDst(importDirectoryDst / newFilename);
+            AZ::IO::Path targetPathAssetTmp(importDirectoryTmp / newFilename);
 
             AZ::IO::Path targetPathAssetInfo(targetPathAssetDst.Native() + ".assetinfo");
 
             if (!fileIO->Exists(targetPathAssetDst.c_str()))
             {
-                const auto outcomeMoveDst = fileIO->Copy(resolvedPath.c_str(), targetPathAssetDst.c_str());
+                const auto outcomeCopyTmp = fileIO->Copy(resolvedPath.c_str(), targetPathAssetTmp.c_str());
                 AZ_Printf(
                     "CopyAssetForURDF",
                     "Copy %s to %s, result: %d",
                     resolvedPath.c_str(),
-                    targetPathAssetDst.c_str(),
-                    outcomeMoveDst.GetResultCode());
-                if (outcomeMoveDst)
+                    targetPathAssetTmp.c_str(),
+                    outcomeCopyTmp.GetResultCode());
+                if (outcomeCopyTmp)
                 {
-                    copiedFiles[unresolvedUrfFileName] = targetPathAssetDst.String();
+
+                    // create asset info at destination
+                    const bool assetInfoOk =
+                        CreateSceneManifest(targetPathAssetTmp.String(), targetPathAssetInfo.String(), needsCollider, needsVisual);
+
+                    if (assetInfoOk)
+                    {
+                        const auto outcomeMoveDst = fileIO->Copy(targetPathAssetTmp.c_str(), targetPathAssetDst.c_str());
+                        AZ_Printf(
+                            "CopyAssetForURDF",
+                            "Move %s to %s, result: %d",
+                            targetPathAssetTmp.c_str(),
+                            targetPathAssetDst.c_str(),
+                            outcomeMoveDst.GetResultCode());
```

Comment:
Actually for speed and atomicity, it would be better to use the [FileIOBase::Rename](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/FileIO.h#L157-L160) function here to perform the equivalent of a `mv` command on.

That way there is no danger of the AP pickup a partially cppied assetinfo file as it is being copied.

Reply:
Great suggestion! Thanks!